dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+    correlator : str
+        Source of the input data
+    acq_suffix : str
+        Type of dataset to be read
+    source_transits : list of str
+        Source tranits that we want analyses for
+    output_suffix : str
+        Suffix for the output file
+    cyl_start_char : int
+        Starting character for the cylinders (ASCII)
+    cyl_start_num : int
+        Offset for CHIME cylinders
+        (due to inital numbers allotted to pathfinder)
+    sep_cyl : bool
+        option to preserve cylinder pairs
+    include_auto : bool
+        option to include autocorrelation
+    include_intracyl : bool
+        option in include intracylinder baselines
+    include_crosspol : bool
+        option to include crosspol data
+    rot : float
+        Rotation angle of the cylinder in degrees
+    nfreq_per_block : int
+        number of frequency channels to be run in one block
+        loading all frequency channels at the same time leads to memory error
+    nsgima : float
+        Span required from the peak of transit
+        in terms of sigma
+    perform_fit : boolean
+        Performs a Gaussian fit to transit
+
+    """
+
+    # Config parameters related to scheduling
+    lag = config.Property(proptype=str2timedelta, default="4h")
+
+    # Config parameters defining data file selection
+    correlator = config.Property(proptype=str, default="chime")
+    acq_suffix = config.Property(proptype=str, default="stack_corr")
+    source_transits = config.Property(proptype=list, default=["cyg_a"])
+
+    # Config parameters defining output data product
+    output_suffix = config.Property(proptype=str, default="sensitivity")
+
+    # Config parameters related to cylinders
+    cyl_start_char = config.Property(proptype=int, default=65)
+    cyl_start_num = config.Property(proptype=int, default=2)
+    sep_cyl = config.Property(proptype=bool, default=False)
+    rot = config.Property(proptype=float, default=-0.088)
+
+    # Config parameters related to optional inclusions
+    include_auto = config.Property(proptype=bool, default=False)
+    include_intracyl = config.Property(proptype=bool, default=False)
+    include_crosspol = config.Property(proptype=bool, default=False)
+
+    # Config parameters related to the algorithm
+    nfreq_per_block = config.Property(proptype=int, default=16)
+    nsigma = config.Property(proptype=float, default=1.0)
+    perform_fit = config.Property(proptype=bool, default=False)
+
+    def setup(self):
+        """
+        Set up the task.
+
+        Creates metrics.
+        """
+        self.run_metric = self.add_task_metric(
+            "source_analyzer_run_total",
+            "task run counter for specific sources",
+            labelnames=["source"],
+            unit="total",
+        )
+
+        self.logger.info("Sources: {}".format(self.source_transits))
+
+    def run(self):
+        """Run the analyzer.
+
+        Load autocorrelations for a given source transit.
```

Comment:
"Load autocorrelations for a given source transit."  --> "Load stacked visibilities and beam form to the location of the brightest sources."